File: LeetCode/1486. XOR Operation in an Array/solution.py
"""
54 / 54 test cases passed.
Runtime: 36 ms
Memory Usage: 14.9 MB
"""
from functools import reduce
from operator import xor
class Solution:
    def xorOperation(self, n: int, start: int) -> int:
        return reduce(xor, [start + 2 * i for i in range(n)])

File: LeetCode/1486. XOR Operation in an Array/solution.cpp
/**
 * 54 / 54 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.8 MB 
 */
class Solution {
public:
    int xorOperation(int n, int start) {
        int ans = 0;
        for (int i = 0; i < n; ++ i ) {
            ans ^= start + 2 * i;
        }
        return ans;
    }
};
